### PR TITLE
DO-267: Completed adding edit icon to each detail section which lead …

### PR DESCRIPTION
**Pull request by ayushmalik2110**
…to corresponding form.

### CODE REVIEW

File: src/utils/member/utils.js [hunk 1]
```diff
@@ -765,10 +765,10 @@ export const isFamilyDetailsDataDirty = (formData = [], updatedFormData = []) =>
...
 /**
- * In order to match the Member_Details route, we need to exclude memberId from the pathName
- * this method will return pathname string excluding the memberId
+ * In order to match route, we need to exclude the hash router param from the pathName
+ * this method will return pathname string excluding the router param.
  * @param {String} pathname
  * @return {string}
  */
-export const getMemberDetailsPathFromPathName = pathname =>
+export const getPathFromPathName = pathname =>
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:773]: We should rename  it to "removeUrlParamsFromPathname"

File: src/components/core/GoogleAnalytics/GoogleAnalytics.jsx [hunk 2]
```diff
@@ -18,10 +18,20 @@ class GoogleAnalytics extends React.Component {
...
   componentDidUpdate({ history }) {
     let { location: { pathname } } = history;
-    if (pathname.includes(Routes.MEMBER_DETAILS)) {
-      // In order to match the Member_Details route, we need to exclude memberId from the pathName
-      pathname = getMemberDetailsPathFromPathName(pathname);
+
+    // Here we are checking for route which have params append at it.
+    // If route contains a param then we remove the param. This for
+    // matching the path properly.
+    let isPathWithParam = false;
```
Comment (mohitsaladiya) [inline src/components/core/GoogleAnalytics/GoogleAnalytics.jsx:25]: Can be simplified to:
```suggestion
  // Define this in constant file.
    const urlListWithParams = ['MEMBER_DETAILS_WITH_PARAM', 'MEMBER_PROFILE_WITH_PARAM' ];
    
    if (urlListWithParams.includes(pathname)) {
      pathname = removeUrlParamsFromPathname(pathname);
    }

```

File: src/utils/member/utils.js (hunk 1 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js]: Done

File: src/components/core/GoogleAnalytics/GoogleAnalytics.jsx (hunk 2 above)
Comment (ayushmalik2110) [reply src/components/core/GoogleAnalytics/GoogleAnalytics.jsx]: As per discussion it is done.

**Pull request merged**